### PR TITLE
build(tools): add a Makefile that uses uv

### PR DESCRIPTION
**Pull request by lengau**
- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

-----

### CODE REVIEW

**Review by mattculler — APPROVED** [review]
Love it!

File: pyproject.toml [hunk 1]
```diff
@@ -39,6 +39,21 @@ docs = [
     "sphinx-lint==0.9.1",
 ]
 
+[tool.uv]
```
Comment (dariuszd21) [inline pyproject.toml:42]: Is it possible not to duplicate all the extra requirements here, but rather use ` --extra dev` / `--extra lint` / `--extra types` in the Makefile?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+lint-codespell: ## Check spelling with codespell
+	uv run codespell --toml pyproject.toml $(SOURCES)
+
+.PHONY: lint-docs
+lint-docs:  ## Lint the documentation
+	uv run --extra docs sphinx-lint --max-line-length 80 --enable all $(DOCS)
+
+.PHONY: lint-mypy
+lint-mypy: ## Check types with mypy
+	uv run mypy $(SOURCES)
```
Comment (dariuszd21) [inline Makefile:81]: We're currently using `--install-types` and `--non-interactive`, shouldn't we keep it ?
```suggestion
	uv run mypy --install-types --non-interactive $(SOURCES)
```

File: pyproject.toml (hunk 1 above)
Comment (lengau) [reply pyproject.toml:42]: It's possible, but the long-term goal is to remove those extras and just have them as dev dependencies. We can't currently set internal dependency groups as dev dependencies (see: https://github.com/astral-sh/uv/issues/7033) though.

File: Makefile (hunk 2 above)
Comment (lengau) [reply Makefile:81]: Possibly, but using `--install-types` has ended up with some issues so I'd prefer to require the types to be specified as dev dependencies.

Comment (lengau) [reply Makefile:81]: One major issue with `--install-types` is it depends on a `pip` module in the venv, which we don't get when using `uv` unless we explicitly request it.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ifeq ($(OS),Linux)
+	sudo snap install codespell ruff shellcheck
+	sudo snap install --classic --beta astral-uv
+	sudo snap alias astral-uv.uv uv
+	sudo snap alias astral-uv.uvx uvx
+else ifeq ($(OS),Windows_NT)
+	pipx install uv
+	choco install shellcheck
+else ifeq ($(OS),Darwin)
+	bash -c 'curl -LsSf https://astral.sh/uv/install.sh | sh'
```
[the remark this answers is not in the document]
Comment (lengau) [reply Makefile]: Good idea!

**Review by dariuszd21 — APPROVED** [review]
I'm not a fan of the duplicated dependencies (especially in a tool specific section), but besides that I think it looks pretty cool.
I like the idea of Make-driven approach, but I think if we go that way we should do it the same way as pydantic 2 migration - all tools at one time.
If any library or app will stay behind, we we'll still be left with the problem of inconsistency in our own repositories

**Pull request merged**